crypto/evp/p_lib.c

```diff
@@ -1188,6 +1188,26 @@ static int legacy_asn1_ctrl_to_param(EVP_PKEY *pkey, int op,
             }
             return rv;
         }
+    case ASN1_PKEY_CTRL_SUPPORTS_MD_NID:
+        {
+            const OSSL_PARAM *gettable = NULL;
+            OSSL_PARAM params[2];
+
+            if ((gettable = EVP_PKEY_gettable_params(pkey)) == NULL
```

Comment:
This works but there are other ways of detecting if something was modified.  Look at `OSSL_PARAM_modified()`.